orm/idgen_bucket.go

```diff
@@ -0,0 +1,38 @@
+package orm
+
+import "github.com/iov-one/weave"
+
+type IDGenerator interface {
+	NextVal(db weave.KVStore, obj CloneableData) []byte
+}
+
+type IDGenBucket struct {
+	Bucket
+	idGen IDGenerator
+}
+
+func WithSeqIDGenerator(b Bucket, seqName string) IDGenBucket {
+	seq := b.Sequence(seqName)
+	return WithIDGenerator(b, IDGeneratorFunc(func(db weave.KVStore, _ CloneableData) []byte {
```

Comment:
But this means the same sequence is used for all objects.

Eg, 
Id 1,version 1
Id 2,version 2
Id 1,version 3...

Dont we need to read from the actual object